src/pathwayvis/services/actions/actions.service.ts

```diff
@@ -43,8 +43,8 @@ export class ActionsService {
    * @return {types.Action} Action
    */
   public getAction(type: string): types.Action {
-    let actionListAux = actionsList.filter((action: types.Action) => action.type === type);
-    return ((actionListAux.length > 0) ? actionListAux[0] : []);
+    actionsList = actionsList.filter((action: types.Action) => action.type === type);
+    return ((actionsList.length > 0) ? actionsList[0] : null);
```

Comment:
no need for those parens:

    return actionsList.length > 0 ? actionsList[0] : null;

will do the job. Even

    return actionsList[0] || null;

would work, since an empty list would return `undefined` for `list[0]` which is falsy, so the `||` operator will return the second operand.